src/sentry/scim/endpoints/teams.py

```diff
@@ -0,0 +1,279 @@
+
+    def get(self, request, organization, team):
+        context = serialize(team, serializer=TeamSCIMSerializer())
+        return Response(context)
+
+    def _add_members_operation(self, request, operation, team):
+        for member in operation["value"]:
+            member = OrganizationMember.objects.get(
+                organization=team.organization, id=member["value"]
+            )
+            with transaction.atomic():
+                omt = OrganizationMemberTeam.objects.create(team=team, organizationmember=member)
+                self.create_audit_entry(
+                    request=request,
+                    organization=team.organization,
+                    target_object=omt.id,
+                    target_user=member.user,
+                    event=AuditLogEntryEvent.MEMBER_JOIN_TEAM,
+                    data=omt.get_audit_log_data(),
+                )
+
+    def _remove_members_operation(self, request, operation, team):
+        try:
+            # grab the filter out of the brackets of the string that looks
+            # like so: members[userName eq "baz@sentry.io"]
+            parsed_filter = parse_filter_conditions(
+                re.search(r"\[(.*?)\]", operation["path"]).groups()[0]
+            )
+        except Exception:
+            # TODO: log parse error
+            raise ParseError(detail=SCIM_400_INVALID_FILTER)
+        member = OrganizationMember.objects.get(organization=team.organization, id=parsed_filter[0])
+
+        with transaction.atomic():
+            omt = OrganizationMemberTeam.objects.get(team=team, organizationmember=member)
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=omt.id,
+                target_user=member.user,
+                event=AuditLogEntryEvent.MEMBER_LEAVE_TEAM,
+                data=omt.get_audit_log_data(),
+            )
+            omt.delete()
+
+    def _rename_team_opereation(self, request, operation, team):
+        serializer = TeamSerializer(
+            team,
+            data={
+                "slug": slugify(operation["value"]["displayName"]),
+            },
+            partial=True,
+        )
+        if serializer.is_valid():
+            team = serializer.save()
+            self.create_audit_entry(
+                request=request,
+                organization=team.organization,
+                target_object=team.id,
+                event=AuditLogEntryEvent.TEAM_EDIT,
+                data=team.get_audit_log_data(),
+            )
+
+    def patch(self, request, organization, team):
+        """
+        A SCIM Group PATCH request takes a series of operations to peform on a team.
+        It does them sequentially and if any of them fail no operations should go through.
+        The operations are add members, remove members, replace members, and rename team.
+        """
+        try:
+            with transaction.atomic():
```

Comment:
for these PATCH ops, we want them to all be within the same transaction -- if one of them fails, don't do any of them. I'm not sure if I still need logical groupings of the inner transactions or not, but i added them.

Reply:
Is there a limit on the number of operations they might send through here? I don't like the idea of holding a transaction open for an arbitrary number of ops.

Reply:
So I don't think the spec defines that, another implementation i'm seeing caps it at 100, which seems like a sane limit to me, i'll go ahead and add that. thanks for pointing that out.

Reply:
aha, found in okta's SCIM app submission form:
<img width="880" alt="Screen Shot 2021-06-15 at 8 24 07 PM" src="https://user-images.githubusercontent.com/1976777/122153441-dbac3d80-ce17-11eb-9b24-628bb2e8f672.png">

keeping at 100 should be fine, as they can just send multiple patch requests.